### ISSUE TITLE
[RFC]: Canberra distance between two strided arrays

### ISSUE THREAD

**Issue by Planeshifter** [stdlib-js/stdlib#831]
### Description

This RFC proposes adding a function to calculate the Canberra distance between two strided arrays.

Package: `@stdlib/math/strided/distances/canberra`

The function should have the following signature `canberra( N, x, strideX, y, strideY )`.

The Canberra distance is a measure of dissimilarity between two samples in a data set.

The Canberra distance between two elements is as follows:

```js
(abs(x[i] - y[i]) / (abs(x[i]) + abs(y[i]))
```

See:
-   [SciPy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.canberra.html#scipy.spatial.distance.canberra)
-   [compute-io](https://github.com/compute-io/canberra-distance)

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by saisrikardumpeti** [comment]
working on this.

**Comment by kgryte** [comment]
@the-r3aper7 I suggest working on a different issue first. Implementing this feature will likely be more involved, as requires some understanding of our strided array machinery.

**Comment by saisrikardumpeti** [comment]
ok

**Comment by saisrikardumpeti** [comment]
will it be okay if i learn it first then come to this?

**Comment by kgryte** [comment]
Yes, you can try. I suggest having a good look at `math/strided/special` to get a feel for what is there. The OP needs to be updated, as the proposed signature is not correct.

**Comment by saisrikardumpeti** [comment]
Yes amazing!

**Comment by Witty-31-06** [comment]
Hi @kgryte I would like to work on this but there is a status: Blocked label, is this open?

**Comment by batpigandme** [comment]
@Witty-31-06 It’s still open. Per, Athan’s comments above, the blocker was figuring out the interface for strided arrays. There’s been progress there, so check out the implemented strided functions first.

**Comment by kgryte** [comment]
@Witty-31-06 See the packages in https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/strided/distances.

**Comment by kgryte** [comment]
I've removed the blocked label. To start, I suggest adding `stats/strided/distances/dcanberra`, which should be rather similar to , e.g., `stats/strided/distances/dcityblock`.

**Comment by Witty-31-06** [comment]
Thanks for reference. Will check them out.
